Formula/fstar.rb

```diff
@@ -36,32 +35,40 @@ def install
                                            "$(DATE_EXEC) '+%Y-%m-%dT%H:%M:%S%z'"
 
     resource("z3").stage do
-      system "python", "scripts/mk_make.py", "--prefix=#{libexec}"
-      system "make", "-C", "build"
-      system "make", "-C", "build", "install"
+      # F* warns if the Z3 git hash doesn't match
+      githash = Utils.popen_read("git", "rev-parse", "--short=12", "HEAD").chomp
+      system "python", "scripts/mk_make.py", "--prefix=#{libexec}", "--githash=#{githash}"
+      cd "build" do
+        system "make"
+        system "make", "install"
+      end
     end
 
     system "opam", "init", "--no-setup"
-    inreplace "opamroot/compilers/4.05.0/4.05.0/4.05.0.comp",
+    inreplace "opamroot/compilers/4.06.1/4.06.1/4.06.1.comp",
       '["./configure"', '["./configure" "-no-graph"' # Avoid X11
 
-    system "opam", "switch", "4.05.0"
+    system "opam", "switch", "4.06.1"
```

Comment:
It should be able to use 
```
depends_on "ocaml" => :build
```
instead of the `opam switch`. See `coccinelle`, `neal`, `zero-install` etc. for examples

Reply:
Is there an idiomatic way of specifying minimum and maximum versions?
FStar won't build properly with OCaml < 4.04.0.

Reply:
Nope. That dependency specifically means the latest version of the OCaml formula and nothing else:
https://github.com/Homebrew/homebrew-core/blob/master/Formula/ocaml.rb#L17

Reply:
Then I'd prefer to keep the `opam switch` because there's no guarantee that the formula will build with future versions of OCaml.

Reply:
If it breaks later, we can always put it back.

Reply:
Ok.